modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -341,9 +353,54 @@ public static Result analyze(Query query) {
         };
     }
 
+    static Function<Query, Result> pointRangeQuery() {
+        return query -> {
+            PointRangeQuery pointRangeQuery = (PointRangeQuery) query;
+            // We need to check the number type before indexing to ensure we support it.
+            // hacky, but otherwise we need to use the MapperService to determine what kind of number field this is
+            Class<?> enclosingClass = pointRangeQuery.getClass().getEnclosingClass();
+            final byte[] lowerPoint = pointRangeQuery.getLowerPoint();
+            final byte[] upperPoint = pointRangeQuery.getUpperPoint();
+            final RangeType rangeType;
+            final Number width;
+            if (LongPoint.class.isAssignableFrom(enclosingClass)) {
```

Comment:
I think I would not check the instances' classes but instead compute how many values the interval has using NumericUtils.subtract (it returns a byte[] that is comparable).